kernel/kernel-api/src/test/scala/io/delta/kernel/internal/tablefeatures/TableFeaturesSuite.scala

```diff
@@ -785,7 +800,50 @@ class TableFeaturesSuite extends AnyFunSuite {
         7,
         set("columnMapping", "deletionVectors"),
         set("columnMapping", "icebergCompatV2", "deletionVectors")),
-      set("icebergCompatV2"))).foreach {
+      set("icebergCompatV2")),
+    (
+      testMetadata(tblProps = Map("delta.enableIcebergWriterCompatV1" -> "true")),
+      new Protocol(1, 2),
+      new Protocol(
+        2,
+        7,
+        set(),
+        set(
+          "columnMapping",
+          "appendOnly",
+          "invariants",
+          "icebergCompatV2",
+          "icebergWriterCompatV1")),
+      set("icebergCompatV2", "columnMapping", "icebergWriterCompatV1")),
+    (
+      testMetadata(tblProps = Map(
+        "delta.enableIcebergWriterCompatV1" -> "true",
+        "delta.enableDeletionVectors" -> "true")),
```

Comment:
I know we're not enforcing this yet, but why have a test where dvs are enabled? they will be blocked? won't we just have to change this test?